cdap-api/src/main/java/co/cask/cdap/internal/specification/FormatSpecification.java

```diff
@@ -20,24 +20,30 @@
 import co.cask.cdap.internal.io.Schema;
 import com.google.common.base.Objects;
 
+import java.util.Collections;
 import java.util.Map;
 
 /**
  * Specification for a {@link RecordFormat}, including the class, schema, and settings to use for the format.
  */
 public class FormatSpecification {
-  private final String formatClass;
+  private final String name;
   private final Schema schema;
   private final Map<String, String> settings;
 
-  public FormatSpecification(String formatClass, Schema schema, Map<String, String> settings) {
-    this.formatClass = formatClass;
+  // for Gson deserialization
+  private FormatSpecification() {
```

Comment:
or because you want to have `settings` an empty map if the settings is missing from the json string?